### DISCUSSION TITLE
Eigene Parameter an Geräten und Räumen speichern

### DISCUSSION THREAD

**Discussion by Weseler485** [OpenCCU/OpenCCU#2238] [🇩🇪 Ideen]
Hey Leute,
habe da eine Idee, die mir sehr helfen würde, ich aber nicht weiß ob es ohne die OCCU anzupacken überhaupt möglich ist umzusetzen.

Ich würde gerne an Geräten und Gewerken eigene Felder hinterlegen um dort Werte abzuspeichern, welche ich in Scripten abfragen kann. So wäre es am Broll zB interessant zu hinterlegen, auf welchem Rollolevel das Rollo nur auf Spalt offen ist (ist bei mir bei jedem Rollo anders). Bei Räumen würde ich z.B. gerne hinterlegen um welche Uhrzeit das Rollo frühstens hochgehen soll.

Klar, kann man in Systemvariablen ablegen oder in jedem Script einzeln hinterlegen. Fände ich aber an den Stellen schöner und wenn ihr sagt, dass das technisch überhaupt geht und ihr ggf. noch zwei drei Hinweise für eine Umsetzung habt würde ich mich gerne auf diesen Weg machen...

Danke euch!!
Maik2208

**Comment by Baxxy13** [comment]
Was ich mich frage wie die so hinterlegten Informationen dann weitergenutzt werden sollen.

Metadaten kann man jetzt schon fast überall dranhängen und auch per Script wieder auslesen.
Das ist aber umstritten weil es die homematic.regadom "aufbläht".

**Reply by Weseler485** [reply]
Ich kann immer nur für mich sprechen. Aber um beim Beispiel Broll zu bleiben: Ich habe verschiedene Scripte wo für mich das RolloLevel für "Lücke" interessant ist. Das ist je Broll aber ein anderer Wert. Schlafzimmer 19, Bad 16, Küche 20. Jetzt wäre es für mich echt toll diesen Wert direkt am Gerät zu haben und damit in jedem Script direkt zugänglich. Der aktuelle Weg für jedes Gerät eine Systemvariablen pro zusätzlichem "Datenpunkt" anzulegen macht es wirklich nicht übersichtlicher und einfacher.

Beispiel zwei:
Abhängig vom Sonnenaufgang und diversen anderen Faktoren lasse ich morgens die Rollos fahren. Im Wohnbereich aber zB niemals vor 8 Uhr, im Schlafzimmer niemals vor 9  Uhr. Diese Grenze direkt am jeweiligen Raum zu hinterlegen hätte wieder Charme. Dann hätte ich es für jedes Rolle das diesem Raum zugeordnet ist direkt im Zugriff.


Das war erstmal nur das, wieso ich überhaupt auf die Idee gekommen bin. Sicherlich hätte ich später noch mehr Anwendungsfälle.

**Reply by jens-maus** [reply]
Nutz dafür Systemvariablen und verknüpfe die mit dem Kanal/Gerät. Das ist der UseCase.

**Comment by Baxxy13** [comment]
Beim Beispiel 1 würde ich noch mitgehen. 
Wenn man bei 10 BROLL's je einen Meta-Daten Eintrag mit Text und Wert ablegt erzeugt das wahrscheinlich eine kleinere Blähung der regadom als das Anlegen von 10 SysVars. :wink: 

Aber du kannst auch **eine** TextSysVar anlegen und dir darin eine indexbasierte Werteliste reinschreiben die du dann im Script wieder zerlegst. Beispiel: `"SZ:19;BZ:16;KU:20"` usw.

Beispiel 2 lässt sich doch easy mit den Zeitmodulen in Programmen erschlagen.

**Comment by MichaelN0815** [comment]
Wie Baxxy angedeutet hat, kannst du Meta Daten schreiben und lesen.
